src/ogdf/fileformats/DotLexer.cpp

```diff
@@ -198,25 +198,34 @@ bool Lexer::tokenizeLine()
 }
 
 
-bool Lexer::match(const Token::Type &type)
+bool Lexer::match(const Token::Type &type, const bool word)
 {
-	return match(Token::toString(type));
+	return match(Token::toString(type), word);
 }
 
 
-bool Lexer::match(const std::string &str)
+bool Lexer::match(const std::string &str, const bool word)
 {
+	const size_t strLength = str.length();
 	// Check whether buffer is too short to match.
-	if(m_buffer.length() - m_col < str.length()) {
+	if(m_buffer.length() < m_col + strLength) {
 		return false;
 	}
 
-	for(size_t i = 0; i < str.length(); i++) {
+	for(size_t i = 0; i < strLength; i++) {
 		if(m_buffer[m_col + i] != str[i]) {
 			return false;
 		}
 	}
 
+	// we've matched a part of a word instead of a whole word
+	if (word) {
+		if (m_buffer.length() >= m_col + strLength + 1 &&
+				isalnum(m_buffer[m_col + strLength])) {
```

Comment:
I wonder if `isalnum` is enough. [It seems](https://www.graphviz.org/doc/info/lang.html) that underscores (`_`) and characters from `\200` to `\377` are also possible?

Reply:
Good catch, I'll change that.

Reply:
Has there been something in the old code to check for keywords (including a check for \200–\377 etc)? If so, it could be reused. If not, it's probably nasty...

Reply:
If we interpret the spec to mean that everything in the range from `0200` and `0377` are valid letters, we could just have something like,

`bool isDotAlnum(unsigned char c) { return std::isalnum(c) || c >= 0200 && c <= 0377; }`

Characters `0200` to `0377` fit within a byte so `std::string` should be fine. Otherwise, we'd have to deal with unicode which is a mildly annoying in c++. I think `pugixml` has its own unicode decoder. Since we're using c++11, we can also use [`std::codecvt_utf8`](http://en.cppreference.com/w/cpp/locale/codecvt_utf8).

Reply:
We can probably also use:

```c++
bool isDotAlnum(char c) {
    return isalnum(c) || c < 0;
}
```

Reply:
Yes, I don't care.